sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -503,13 +503,32 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
     }
   }
 
+  private def convertTypeConstructedLiteralToString(literal: Literal): String = literal match {
+    case Literal(data: Int, dataType: DateType) =>
+      UTF8String.fromString(
+        DateFormatter(getZoneId(SQLConf.get.sessionLocalTimeZone))
+          .format(data)).toString
+    case Literal(data: Long, dataType: TimestampType) =>
+      UTF8String.fromString(
+        TimestampFormatter.getFractionFormatter(getZoneId(SQLConf.get.sessionLocalTimeZone))
+          .format(data)).toString
+    case Literal(data: CalendarInterval, dataType: CalendarIntervalType) =>
+      UTF8String.fromString(data.toString).toString
+    case Literal(data: Array[Byte], dataType: BinaryType) =>
+      UTF8String.fromBytes(data).toString
+    case Literal(data, dataType) =>
+      UTF8String.fromString(data.toString).toString
```

Comment:
We need this entry? What's an example to match this case?

Reply:
> We need this entry? What's an example to match this case?

Normally this won't happen, just remove this match case?

Reply:
> We need this entry? What's an example to match this case?

removed this line

Reply:
Add back...Seems I add this case  avoid build problem.
```
[error] /home/jenkins/workspace/SparkPullRequestBuilder@3/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala:506:81: Exhaustivity analysis reached max recursion depth, not all missing cases are reported.
[error] (Please try with scalac -Ypatmat-exhaust-depth 40 or -Ypatmat-exhaust-depth off.)
[error]   private def convertTypeConstructedLiteralToString(literal: Literal): String = literal match {
[error]                                                                                 ^
[error] one error found
```

Reply:
Yea, that's the effect of the recent compiler update: https://github.com/apache/spark/pull/30455

Reply:
how about throwing an illegal state exception in the case?

Reply:
> how about throwing an illegal state exception in the case?

Good suggestion , updated.

> Yea, that's the effect of the recent compiler update: #30455

Thanks for point out   this pr.